Add option to change default eyes when joining server

sjrc6 opened the pull request:
Risky Feat suggested this idea to me, I also thought it was good so I implemented it. 

If you press one of the eyes in the settings it will change which eyes you get when you join a server.

It saves a separate variable for dummy/player.

It will also change your eyes anytime you press one of them, but only if you are playing as the same Tee whose eyes you are trying to change. If someone could figure out the best way to make it change the opposite Tees eyes as well that would be nice but I don't think it's necessary. 

It also changes the "Your Skin:" Tee eyes on the left if you have one selected. 

I had to change the cl_run_on_join code a bit, I tested and I don't think it should ever break anything but it might be good for someone to double check. 

![image](https://user-images.githubusercontent.com/22122579/163762410-ec7aa67d-6417-433d-8707-e21cda2bcd5f.png)



## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

def- commented:
![screenshot-20220418@100129](https://user-images.githubusercontent.com/2335377/163777087-4c34793f-ffb9-42f6-9a5f-5ed57d6880f9.png)
Everything has to fix with 5:4, for UI changes I suggest testing with 5:4 resolution always.

def- commented on src/engine/client/client.cpp:
```diff
@@ -2103,13 +2103,41 @@ void CClient::ProcessServerPacket(CNetChunk *pPacket, int Conn, bool Dummy)
+								char aBufEmote[256];
+								switch(Emote)
+								{
+								case EMOTE_NORMAL:
```
Let's just send nothing for the default case? Otherwise this feature destroys the server-side halloween feature of giving players angry eyes by default.

sjrc6 replied:
Yes that's fair I didn't think about that

sjrc6 commented:
I fixed it, also removed a lot of code because it's in a loop now.

If you think it's excessive to render all 6 options as a fancy button we could also just make it a single text button that cycles when you press it. But I like how it doubles as a way to view all the eyes for the skin you select. 

![image](https://user-images.githubusercontent.com/22122579/163845707-5212aa95-41aa-4d0d-93d7-34a7b4e011a4.png)

![image](https://user-images.githubusercontent.com/22122579/163846284-71fc6bb9-ab1b-40a7-8a38-cb8e53986c79.png)

def- commented:
I think it's ok. Other opinions?

srdante commented:
I think the padding around those icons is not being considered. The 4:3 is currently tight with elements on top and bottom, and the 16:9 is not respecting the right padding. Example of 16:9:

![image](https://user-images.githubusercontent.com/35383529/163864006-3fb0e307-79ff-4a07-bf25-a034fe6ffb21.png)

sjrc6 commented:
It respects the right side padding now 

![image](https://user-images.githubusercontent.com/22122579/163870993-4a00a002-fb4a-4372-aad7-e43d9845c157.png)
![image](https://user-images.githubusercontent.com/22122579/163871026-3845e192-547d-498c-a8cd-5752dadc8ea6.png)
![image](https://user-images.githubusercontent.com/22122579/163871121-efae2563-d2fd-4dfb-a138-187b6f8f4471.png)

murpii commented:
> I think it's ok. Other opinions?

I mean.. isn't this already possible with cl_run_on_join ? 

> it will also change your eyes anytime you press one of them

The emote wheel has all eyes aswell

heinrich5991 commented on src/game/variables.h:
```diff
@@ -104,6 +104,7 @@ MACRO_CONFIG_INT(ClPlayerUseCustomColor, player_use_custom_color, 0, 0, 1, CFGFL
 MACRO_CONFIG_COL(ClPlayerColorBody, player_color_body, 65408, CFGFLAG_CLIENT | CFGFLAG_SAVE | CFGFLAG_COLLIGHT | CFGFLAG_INSENSITIVE, "Player body color")
 MACRO_CONFIG_COL(ClPlayerColorFeet, player_color_feet, 65408, CFGFLAG_CLIENT | CFGFLAG_SAVE | CFGFLAG_COLLIGHT | CFGFLAG_INSENSITIVE, "Player feet color")
 MACRO_CONFIG_STR(ClPlayerSkin, player_skin, 24, "default", CFGFLAG_CLIENT | CFGFLAG_SAVE | CFGFLAG_INSENSITIVE, "Player skin")
+MACRO_CONFIG_INT(ClPlayerDefaultEyes, player_defualt_eyes, 0, 0, 5, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Player eyes when joining server")
```
```suggestion
MACRO_CONFIG_INT(ClPlayerDefaultEyes, player_default_eyes, 0, 0, 5, CFGFLAG_CLIENT | CFGFLAG_SAVE, "Player eyes when joining server")
```
Typo.

heinrich5991 replied:
I'd also list the possible choices here.

heinrich5991 reviewed (commented):
LGTM otherwise.

C0D3D3V commented:
I had another idea. You could do this as a popup. You could make the "Your Skin" look like a button, with tooltip "click to select the eyes your tee has by default when joining a server". And then when you click on it a popup opens with the 6 choices.

sjrc6 commented:
I think it's good enough now. I fixed the typo, added some more vertical padding when rendering below, and added a tooltip. 

I had to change the render order to prevent the tooltip from rendering under the tees but that actually ended up being less code than before so that's fine. 

Tool tips changed since I started this pr so to get it to merge the branch doesn't compile by itself, I tested it locally and it works fine though. probably there is some obscure git technique to fix this idk

def- commented:
> Tool tips changed since I started this pr so to get it to merge the branch doesn't compile by itself, I tested it locally and it works fine though. probably there is some obscure git technique to fix this idk

Not so obscure, but basic git functionality: merge or rebase depending on your taste. I can do that.

sjrc6 commented:
> The tooltip activation feels weird to me. Sometimes it doesn't activate, sometimes I move my mouse a bit and it deactivates again, sometimes not. I'm not seeing this behavior with other tooltips.

Is it because the tooltip doesn't activate between the buttons?

sjrc6 commented:
rebase worked thanks, I was overthinking it

def- commented:
No, it happens even when moving the cursor a bit on the same button

sjrc6 commented:
hmm let me see

def- commented:
Checked again and it only happens after I resized window. I think the tooltips don't handle that properly, I'll open a new issue for this, unrelated to this change.

sjrc6 commented:
it seems like changing resolution after starting the game breaks tooltips?

def- commented:
Yep, opened bug: https://github.com/ddnet/ddnet/issues/5026

def- commented on src/engine/client/client.cpp:
```diff
@@ -2103,13 +2103,40 @@ void CClient::ProcessServerPacket(CNetChunk *pPacket, int Conn, bool Dummy)
+									str_format(aBufEmote, sizeof(aBufEmote), "emote blink %d", g_Config.m_ClEyeDuration);
+									break;
+								}
+								str_format(aBuf, sizeof(aBuf), "%s;%s", aBuf, aBufEmote);
```
Minor comment: For the EMOTE_NORMAL case would be nice not to insert `;`. Maybe just an `if(aBufEmote[0])` around this.

def- commented on src/engine/client/client.cpp:
```diff
@@ -2103,13 +2103,40 @@ void CClient::ProcessServerPacket(CNetChunk *pPacket, int Conn, bool Dummy)
 							{
-								str_format(aBuf, sizeof(aBuf), "/timeout %s", m_aTimeoutCodes[Conn]);
+								int Emote = ((g_Config.m_ClDummy) ? !Dummy : Dummy) ? g_Config.m_ClDummyDefaultEyes : g_Config.m_ClPlayerDefaultEyes;
+								char aBufEmote[128];
```
`aBufEmote[0] = '\0'`
Otherwise any old memory can still be in here with EMOTE_NORMAL.

sjrc6 commented:
@def- I think I addressed those changes
